app/models/amusement_park.rb

```diff
@@ -1,3 +1,8 @@
 class AmusementPark < ApplicationRecord
   has_many :rides
+  has_many :mechanics, through: :rides
+
+  def mechanic_staff
+    mechanics.distinct
```

Comment:
Awesome!! This story is so much easier with the additional association with mechanics